### ISSUE TITLE
Specifying histtype='stepfilled' and normed=True when using plt.hist causes ymax to be set incorrectly

### ISSUE THREAD

**Issue by joelostblom** [matplotlib/matplotlib#4414]
The combination of `histtype='stepfilled'` and `normed=True` seem to cause some problems with the max ylimit.

``` python
import matplotlib.pyplot as plt
import seaborn.apionly as sns
iris = sns.load_dataset('iris')

plt.figure()
for species, species_df in iris.groupby("species"):
    plt.hist(species_df["petal_length"].values, normed=True, histtype="stepfilled")
plt.legend(sorted(iris.species.unique()))
```

![image](https://cloud.githubusercontent.com/assets/4560057/7525297/230b7ae6-f4d6-11e4-8738-5399fdd6f88f.png)
Which is not the case when only using `histtype='stepfilled'`.

``` python
plt.figure()
for species, species_df in iris.groupby("species"):
    plt.hist(species_df["petal_length"].values, histtype="stepfilled")
```

![image](https://cloud.githubusercontent.com/assets/4560057/7525309/3ed7bd66-f4d6-11e4-8f8a-e54677e7cc17.png)
Or when only using `normed=True`.

``` python
plt.figure()
for species, species_df in iris.groupby("species"):
    plt.hist(species_df["petal_length"].values, normed=True)
```

![image](https://cloud.githubusercontent.com/assets/4560057/7525319/53cfd0aa-f4d6-11e4-99ed-88667cea5db4.png)
It seems like in the erroneous case above, the max of the yaxis is set by the last plotted distribution, which can be tested by changing the order of when the different histograms are plotted.

``` python
plt.figure()
iris.loc[iris.species == 'setosa', 'species'] = 'wsetosa'
for species, species_df in iris.groupby("species"):
    plt.hist(species_df["petal_length"].values, normed=True, histtype="stepfilled")
plt.legend(sorted(iris.species.unique()))
```

![image](https://cloud.githubusercontent.com/assets/4560057/7525328/6db22428-f4d6-11e4-8c03-84f65d4f9e72.png)

**Comment by tacaswell** [comment]
I suspect that some place there is a `ax.set_ylim(max(..))` call which just needs to be changed to an autoscale/limit call, but given that this is `hist` one can never be sure.

**Comment by QuLogic** [comment]
This was corrected by #7476 and should be fixed in 2.0.0.
![figure_1](https://cloud.githubusercontent.com/assets/302469/21077451/997aae04-bf19-11e6-83bf-491140b10d98.png)
